server/events/project_command_builder.go

```diff
@@ -217,7 +225,7 @@ func (p *DefaultProjectCommandBuilder) BuildAutoplanCommands(ctx *command.Contex
 // See ProjectCommandBuilder.BuildPlanCommands.
 func (p *DefaultProjectCommandBuilder) BuildPlanCommands(ctx *command.Context, cmd *CommentCommand) ([]command.ProjectContext, error) {
 	if !cmd.IsForSpecificProject() {
-		return p.buildAllCommandsByCfg(ctx, cmd.CommandName(), cmd.Flags, cmd.Verbose)
+		return p.buildAllCommandsByCfg(ctx, cmd.CommandName(), cmd.SubName, cmd.Flags, cmd.Verbose)
```

Comment:
This probably came up before, but why not just pass the whole `cmd` instead of select `SubName`, `Flags`, and `Verbose` ?

Reply:
`BuildAutoplanCommands` doesn't have `cmd *CommentCommand`. We can do like following, but I think this should not be introduced. Because `cmd := CommentCommand{Name: command.Plan}` might be cause nil pointer panic or invalid implementations by contributers who want to change `buildAllCommandsByCfg` and don't know passed lacked  `CommentCommand`.

```diff
>  gd
diff --git a/server/events/project_command_builder.go b/server/events/project_command_builder.go
index d1068da2..08928d88 100644
--- a/server/events/project_command_builder.go
+++ b/server/events/project_command_builder.go
@@ -207,7 +207,8 @@ type DefaultProjectCommandBuilder struct {

 // See ProjectCommandBuilder.BuildAutoplanCommands.
 func (p *DefaultProjectCommandBuilder) BuildAutoplanCommands(ctx *command.Context) ([]command.ProjectContext, error) {
-       projCtxs, err := p.buildAllCommandsByCfg(ctx, command.Plan, "", nil, false)
+       cmd := CommentCommand{Name: command.Plan}
+       projCtxs, err := p.buildAllCommandsByCfg(ctx, cmd)
        if err != nil {
                return nil, err
        }
@@ -225,7 +226,7 @@ func (p *DefaultProjectCommandBuilder) BuildAutoplanCommands(ctx *command.Contex
 // See ProjectCommandBuilder.BuildPlanCommands.
 func (p *DefaultProjectCommandBuilder) BuildPlanCommands(ctx *command.Context, cmd *CommentCommand) ([]command.ProjectContext, error) {
        if !cmd.IsForSpecificProject() {
-               return p.buildAllCommandsByCfg(ctx, cmd.CommandName(), cmd.SubName, cmd.Flags, cmd.Verbose)
+               return p.buildAllCommandsByCfg(ctx, cmd)
        }
```